### ISSUE TITLE
3.5.1 Invalid Broccoli2 node detected

### ISSUE THREAD

**Issue by ihomp** [ember-cli/ember-cli#8198]
WARNING: Invalid Broccoli2 node detected, falling back to broccoli-builder. Broccoli error:
---------------
TemplateCompiler: The .read/.rebuild API is no longer supported as of Broccoli 1.0. Plugins must now derive from broccoli-plugin. https://github.com/broccolijs/broccoli/blob/master/docs/broccoli-1-0-plugin-api.md
used as input node to broccoli-persistent-filter:TemplateCompiler
-~- created here: -~-
    at TemplateCompiler.Plugin (/node_modules/broccoli-plugin/index.js:7:31)
    at new Filter (/node_modules/broccoli-persistent-filter/index.js:78:10)
    at new TemplateCompiler (/node_modules/ember-cli-htmlbars/index.js:32:5)
    at Object.toTree (/node_modules/ember-cli-htmlbars/ember-addon-main.js:46:16)
    at /node_modules/ember-cli-preprocess-registry/preprocessors.js:180:26
    at Array.forEach (<anonymous>)
    at processPlugins (/node_modules/ember-cli-preprocess-registry/preprocessors.js:178:11)
    at module.exports.preprocessTemplates (/node_modules/ember-cli-preprocess-registry/preprocessors.js:161:10)
    at DefaultPackager.processTemplates (/node_modules/ember-cli/lib/broccoli/default-packager.js:467:9)
    at DefaultPackager.processAppAndDependencies (/node_modules/ember-cli/lib/broccoli/default-packager.js:240:28)
-~- (end) -~----------------

[API] Warning: The .read and .rebuild APIs will stop working in the next Broccoli version
[API] Warning: Use broccoli-plugin instead: https://github.com/broccolijs/broccoli-plugin
[API] Warning: Plugin uses .read/.rebuild API: TemplateCompiler
[API] Warning: Plugin uses .read/.rebuild API: ember-browserify
[API] Warning: Plugin uses .read/.rebuild API: ember-browserify

Output from `ember version --verbose && npm --version && yarn --version`:
```
ember-cli: 3.5.1
http_parser: 2.8.0
node: 10.4.1
v8: 6.7.288.45-node.7
uv: 1.20.3
zlib: 1.2.11
ares: 1.14.0
modules: 64
nghttp2: 1.32.0
napi: 3
openssl: 1.1.0h
icu: 61.1
unicode: 10.0
cldr: 33.0
tz: 2018c
os: darwin x64
6.4.1
1.12.1
```

```
"devDependencies": {
    "@ember/jquery": "^0.5.2",
    "@ember/optional-features": "^0.7.0",
    "awesome-phonenumber": "^2.2.6",
    "broccoli-asset-rev": "^3.0.0",
    "ember-ajax": "^4.0.0",
    "ember-body-class": "^1.1.3",
    "ember-browserify": "^1.2.2",
    "ember-cli": "~3.5.1",
    "ember-cli-app-version": "^3.2.0",
    "ember-cli-babel": "^7.1.3",
    "ember-cli-chartjs": "^3.0.0",
    "ember-cli-clipboard": "^0.10.0",
    "ember-cli-content-security-policy": "^1.0.0",
    "ember-cli-data-export-with-style": "^3.1.0",
    "ember-cli-dependency-checker": "^3.0.0",
    "ember-cli-emblem": "0.3.1",
    "ember-cli-eslint": "^4.2.3",
    "ember-cli-google-analytics": "^1.5.0",
    "ember-cli-htmlbars": "^3.0.1",
    "ember-cli-htmlbars-inline-precompile": "^2.0.0",
    "ember-cli-ic-ajax": "^1.0.0",
    "ember-cli-inject-live-reload": "^2.0.1",
    "ember-cli-loading-slider": "^2.0.0",
    "ember-cli-mirage": "^0.4.9",
    "ember-cli-moment-shim": "^3.7.1",
    "ember-cli-password-strength": "^2.0.0",
    "ember-cli-release": "^0.2.9",
    "ember-cli-shims": "^1.2.0",
    "ember-cli-sri": "^2.1.1",
    "ember-cli-stylus": "^1.0.7",
    "ember-cli-template-lint": "^1.0.0-beta.2",
    "ember-cli-test-loader": "^2.2.0",
    "ember-cli-uglify": "^2.1.0",
    "ember-click-outside": "^1.0.2",
    "ember-data": "~3.5.0",
    "ember-data-url-templates": "^0.4.3",
    "ember-disable-proxy-controllers": "^1.0.1",
    "ember-export-application-global": "^2.0.0",
    "ember-i18n": "5.2.1",
    "ember-i18n-inject": "1.0.0",
    "ember-infinity": "1.0.2",
    "ember-inflector": "^3.0.0",
    "ember-intercom-io": "^1.0.1",
    "ember-link-action": "^0.1.2",
    "ember-load-initializers": "^2.0.0",
    "ember-maybe-import-regenerator": "^0.1.6",
    "ember-moment": "^7.8.0",
    "ember-qunit": "^4.1.2",
    "ember-radio-buttons": "^4.1.0",
    "ember-resolver": "^5.0.1",
    "ember-source": "~3.5.1",
    "ember-truth-helpers": "^2.1.0",
    "ember-websockets": "^7.2.0",
    "ember-welcome-page": "^3.2.0",
    "eslint-plugin-ember": "^5.2.0",
    "liquid-fire": "^0.29.5",
    "loader.js": "^4.7.0",
    "qunit-dom": "^0.8.0"
  },
  "engines": {
    "node": "6.* || 8.* || >= 10.*",
    "yarn": ">= 1.0.0"
  },
  "dependencies": {
    "@bower_components/animate.css": "daneden/animate.css#~3.5.0",
    "@bower_components/lodash": "lodash/lodash#~3.7.0",
    "@bower_components/nib": "visionmedia/nib#~1.1.0",
    "@bower_components/route-recognizer": "tildeio/route-recognizer#~0.1.1",
    "@bower_components/sweetalert": "t4t5/sweetalert#~1.1.3",
    "@bower_components/uri.js": "medialize/URI.js#~1.17.0",
    "@bower_components/zxcvbn": "dropbox/zxcvbn#~4.2.0"
  }
```

**Comment by rwjblue** [comment]
Hmm, I'm not sure I understand what you are asking here...

**Comment by ihomp** [comment]
I can't find what is causing the warning @rwjblue 

`Plugins must now derive from broccoli-plugin.`

**Comment by Turbo87** [comment]
```
[API] Warning: The .read and .rebuild APIs will stop working in the next Broccoli version
[API] Warning: Use broccoli-plugin instead: https://github.com/broccolijs/broccoli-plugin
[API] Warning: Plugin uses .read/.rebuild API: TemplateCompiler
[API] Warning: Plugin uses .read/.rebuild API: ember-browserify
[API] Warning: Plugin uses .read/.rebuild API: ember-browserify
```
you are apparently using some Ember addons that are not using the new Broccoli APIs. judging from the stacktrace one of them seems to be `ember-cli-htmlbars` and the other `ember-browserify`.

**Comment by ihomp** [comment]
@Turbo87 thanks, removing unsupported ember-browserify removed two warnings:
[API] Warning: Plugin uses .read/.rebuild API: ember-browserify
[API] Warning: Plugin uses .read/.rebuild API: ember-browserify

but "ember-cli-htmlbars": "^3.0.1", is supported and even recently updated by @rwjblue

**Comment by Turbo87** [comment]
@ihomp it is possible that one of your addons is using an older version of `ember-cli-htmlbars` which is causing this warning. if you use a lockfile make sure to update *all* instances of `ember-cli-htmlbars`.

**Comment by Turbo87** [comment]
idk what the exact problem is here, but `ember-cli-htmlbars` relies on `broccoli-persistent-filter`, and maybe you're using an older version of that somewhere. did you check if the issues is resolved if you ignore your lockfile?

**Comment by ihomp** [comment]
warrnings are stays without a lockfile :( 

I found where broccoli-plugin "^1.0.0" is used.

broccoli-persistent-filter@^1.1.5, broccoli-persistent-filter@^1.1.6, broccoli-persistent-filter@^1.4.3:
  version "1.4.6"
    broccoli-plugin "^1.0.0"

broccoli-persistent-filter@^2.1.1:
  version "2.1.1"
    broccoli-plugin "^1.0.0"


broccoli-persistent-filter used in:

broccoli-stew@^1.4.2, broccoli-stew@^1.5.0:
  version "1.6.0"
    broccoli-persistent-filter "^1.1.6"
    broccoli-plugin "^1.3.0"

broccoli-stew@^2.0.0:
  version "2.0.1"
    broccoli-persistent-filter "^2.1.1"
    broccoli-plugin "^1.3.1"

broccoli-string-replace@^0.1.1, broccoli-string-replace@^0.1.2:
  version "0.1.2"
    broccoli-persistent-filter "^1.1.5"

ember-cli-htmlbars@3.0.0:
  version "3.0.0"
    broccoli-persistent-filter "^1.4.3"

ember-cli-htmlbars@^2.0.1, ember-cli-htmlbars@^2.0.2, ember-cli-htmlbars@^2.0.3:
  version "2.0.5"
    broccoli-persistent-filter "^1.4.3"

ember-cli-htmlbars@^3.0.0, ember-cli-htmlbars@^3.0.1:
  version "3.0.1"
    broccoli-persistent-filter "^1.4.3"

ember-cli-template-lint@^1.0.0-beta.2:
  version "1.0.0-beta.2"
    broccoli-persistent-filter "^1.4.3"

broccoli-asset-rev@^3.0.0:
  version "3.0.0"
    broccoli-persistent-filter "^1.4.3"

broccoli-babel-transpiler@^6.4.5, broccoli-babel-transpiler@^6.5.0:
  version "6.5.0"
    broccoli-persistent-filter "^1.4.3"

broccoli-babel-transpiler@^7.1.0:
  version "7.1.0"
    broccoli-persistent-filter "^1.4.3"

broccoli-clean-css@^1.1.0:
  version "1.1.0"
    broccoli-persistent-filter "^1.1.6"

broccoli-lint-eslint@^4.2.1:
  version "4.2.1"
    broccoli-persistent-filter "^1.4.3"

**Comment by Turbo87** [comment]
hmm, interesting. since both already use `broccoli-plugin` this does not seem to be the problem. at this point I'd try to search the `node_modules` folder for `TemplateCompiler` and see in what packages it is implemented. maybe that gives a clue what the problem is.

**Comment by fran-worley** [comment]
I had this issue, tracked it down in the end to a dependency (ember-simple-auth v1.6.0) restricting ember-fetch to v3. ember-fetch will cause this error < v5 due to reliance on broccoli-templater v1.

**Comment by Turbo87** [comment]
@fran-worley you can fix this by updating to `ember-simple-auth@1.7.0`, which supports newer `ember-fetch` versions too

**Comment by ihomp** [comment]
@Turbo87 Thank you.
I've searched the `node_modules`, found those:

ember-cli
ember-cli-htmlbars-inline-precompile
ember-template-lint
ember-source
ember-resolver
@glimmer
@ember
ember-cli-emblem
liquid-fire
ember-welcome-page
ember-cli-clipboard
ember-cli-chartjs
ember-cli-htmlbars
ember-cli-data-export-with-style
ember-click-outside
ember-cli-preprocess-registry
ember-cli-babel
broccoli-babel-transpiler

The first thing I've checked was `ember-cli-emblem`
Removing it from `package.json` removed the warning for me. 

I've updated ember-cli-emblem, now it works well. 
"ember-cli-emblem": "^0.10.0",

Thank you very much @Turbo87
